cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTableNotFoundException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.data;
+
+import co.cask.cdap.spi.data.table.StructuredTableId;
+
+/**
+ * Exception thrown if the structured table is not found when instantiating the table.
+ */
+public class StructuredTableNotFoundException extends StructuredTableInstantiationException {
+  public StructuredTableNotFoundException(StructuredTableId tableId, String message, Throwable cause) {
```

Comment:
So if a table already exists, we throw AlreadyExistsException. But in this case we don't throw a NotFoundException?

Reply:
Removed this class and throw `NotFondException` to `StructuredTableContext`